doc/rfc/0000-rfc-repository.md

```diff
@@ -0,0 +1,168 @@
+# RFC 0000 - RFC repository
+
+- Date proposed: 2023-02-28
+- Date approved: under review
+
+## Summary
+
+This RFC proposes storing OpenTitan RFCs as files in a Git repo, and using
+an open source development workflow for the RFC process. RFCs would be
+stored in the RFC repo as text files. RFCs would be proposed, reviewed, and
+accepted/rejected using GitHub pull requests (PRs). The goal is to ensure
+project decisions are better documented, and to make the RFC process more
+visible and transparent. The workflow proposed here will already be familiar to
+open source contributors and should add little overhead.
+
+## Motivation
+
+RFCs in OpenTitan are used for decisions that need consensus from the technical
+committee. A summary of the RFC process as documented [here][rfc-process]:
+
+1. Write up your RFC somewhere.
+2. Create an issue in the OpenTitan repo with a link to your RFC.
+3. Label the issue with `RFC:Proposal`, then `RFC:TC review` when ready.
+4. Pending RFCs will be discussed in the next technical committee meeting.
+5. Approved RFCs have their label changed to `RFC:Approved`.
+6. Rejected RFCs have their label removed.
+
+Typically, authors write their RFCs in one of two places:
+
+* A Google Doc, accessible only to select people.
+* The GitHub issue itself as Markdown.
+
+RFCs stored like this are not discoverable and not portable. In the case
+of Google Docs, they are not accessible to everybody, even within the
+opentitan.org domain. RFCs are reviewed in private by the technical committee
+and project director.
```

Comment:
This description seems to mix a lot of different concerns, the way it's currently phrased.

The discovery issue seems mainly related to the lack of a centralized entry point that lists all of the RFCs, not so much to how they are stored, per se.

The permissions issue is also not related to the platform. For instance, having a private GitHub RFC repo wouldn't address those concerns, while ensuring that all of the RFC Google Docs have their permissions set to publicly viewable would.

It's also, arguably, a narrow view of the current process. AFAIK, the final review by the TC (and the potential, rare, project director veto) was not intended as an impediment to collaboration and public iteration of RFCs, before being considered by the TC.

Reply:
I've expanded on the process to include that contributors comment on RFCs before the TC take the final vote.

I've tried to change just saying RFCs aren't discoverable to instead describing how you would currently discover RFCs and why that isn't ideal.

And I've changed the discussion of the permissions issue to be less about Google Docs and more about not having a single platform where we can apply a consistent rule on visibility.